pkg/booleanpolicy/default_policies_test.go

```diff
@@ -2192,52 +2192,52 @@ func (suite *DefaultPoliciesTestSuite) TestImageVerified() {
 	}{
 		{
 			values:          []string{verifier0},
-			negate:          false,
+			negate:          true,
 			expectedMatches: set.NewFrozenStringSet("verified_by_0"),
 		},
 		{
 			values:          []string{verifier0},
-			negate:          true,
+			negate:          false,
 			expectedMatches: allImages.Difference(set.NewFrozenStringSet("verified_by_0")),
 		},
 		{
 			values:          []string{verifier1},
-			negate:          false,
+			negate:          true,
 			expectedMatches: set.NewFrozenStringSet(),
 		},
 		{
 			values:          []string{verifier1},
-			negate:          true,
+			negate:          false,
 			expectedMatches: allImages,
 		},
 		{
 			values:          []string{verifier2},
-			negate:          false,
+			negate:          true,
 			expectedMatches: set.NewFrozenStringSet("verified_by_2_and_3"),
 		},
 		{
 			values:          []string{verifier2},
-			negate:          true,
+			negate:          false,
 			expectedMatches: allImages,
 		},
 		{
 			values:          []string{verifier0, verifier2},
-			negate:          false,
+			negate:          true,
 			expectedMatches: set.NewFrozenStringSet("verified_by_0", "verified_by_2_and_3"),
 		},
 		{
 			values:          []string{verifier0, verifier2},
-			negate:          true,
+			negate:          false,
 			expectedMatches: allImages.Difference(set.NewFrozenStringSet("verified_by_0")),
 		},
 		{
 			values:          []string{verifier3},
-			negate:          true,
+			negate:          false,
```

Comment:
Hm, I'm not really sure I follow that test case's values.
If we have a policy that should alert on all images which are not verifiable by `verifier3`, we mistakenly create an alert for `verified_by_2_and_3`, although this one is in fact verifiable by `verifier3`.
Am I missing something here?

Reply:
That's a very good catch. The engine apparently matches images verified by the exact set of given values, despite the provided OR operator.